crates/wasi/src/instantiate.rs

```diff
@@ -9,9 +9,23 @@ use std::collections::HashMap;
 use std::fs::File;
 use target_lexicon::HOST;
 use wasi_common::{WasiCtx, WasiCtxBuilder};
+use wasmtime_api as api;
 use wasmtime_environ::{translate_signature, Export, Module};
 use wasmtime_runtime::{Imports, InstanceHandle, InstantiationError, VMFunctionBody};
 
+/// Creates `api::Instance` object implementing the "wasi" interface.
+pub fn create_wasi_instance(
+    store: &api::HostRef<api::Store>,
+    preopened_dirs: &[(String, File)],
+    argv: &[String],
+    environ: &[(String, String)],
+) -> Result<api::Instance, InstantiationError> {
+    let global_exports = store.borrow().global_exports().clone();
+    let wasi = instantiate_wasi("", global_exports, preopened_dirs, argv, environ)?;
+    let instance = api::Instance::from_handle(&store, wasi).expect("wasi instance");
```

Comment:
If `from_handle` returns an error, this change makes it panic, where it would have propagated the error before. Should this do `?` instead of `.expect`?